Gofer.NET.Utils/ActionExtensionMethods.cs

```diff
@@ -11,50 +12,50 @@ public static TaskInfo ToTaskInfo(this Expression<Action> expression)
         {
             var methodCallArgumentResolutionVisitor = new MethodCallArgumentResolutionVisitor();
             var expressionWithArgumentsResolved =
-                (Expression<Action>) methodCallArgumentResolutionVisitor.Visit(expression);
+                (Expression<Action>)methodCallArgumentResolutionVisitor.Visit(expression);
 
-            var method = ((MethodCallExpression) expressionWithArgumentsResolved.Body);
+            var method = ((MethodCallExpression)expressionWithArgumentsResolved.Body);
             var m = method.Method;
             var args = method.Arguments
                 .Select(a =>
                 {
-                    var value = ((ConstantExpression) a).Value;
-                    return value;
+                    var value = ((ConstantExpression)a).Value;
+                    return value is CancellationToken ? null : value;
                 })
                 .ToArray();
 
             var taskInfo = m.ToTaskInfo(args);
 
             return taskInfo;
         }
-        
+
         public static TaskInfo ToTaskInfo<T>(this Expression<Func<T>> expression)
         {
             var methodCallArgumentResolutionVisitor = new MethodCallArgumentResolutionVisitor();
             var expressionWithArgumentsResolved =
-                (Expression<Func<T>>) methodCallArgumentResolutionVisitor.Visit(expression);
+                (Expression<Func<T>>)methodCallArgumentResolutionVisitor.Visit(expression);
 
-            var method = ((MethodCallExpression) expressionWithArgumentsResolved.Body);
+            var method = ((MethodCallExpression)expressionWithArgumentsResolved.Body);
             var m = method.Method;
             var args = method.Arguments
                 .Select(a =>
                 {
-                    var value = ((ConstantExpression) a).Value;
-                    return value;
+                    var value = ((ConstantExpression)a).Value;
+                    return value is CancellationToken ? null : value;
```

Comment:
Same thing here regarding the special case.

Reply:
Done